### PR TITLE
Set up web3 with node url argument

### PR DESCRIPTION
**Pull request by e00E**
We need a Web3 instance to make calls to the contract.
Next step is to set up the contracts instances with this web3.

### Test Plan
`cargo run --bin solver -- --node-url "http://localhost:8080"` to see that web3 is created. Doesn't actually need to have a node running at that address because no call is made.

### CODE REVIEW

File: shared-arguments/src/lib.rs [hunk 1]
```diff
@@ -10,6 +11,19 @@ pub struct Arguments {
         default_value = "warn,orderbook=debug,solver=debug"
     )]
     pub log_filter: String,
+
+    /// The Ethereum node URL to connect to.
+    #[structopt(long, env = "NODE_URL")]
```
Comment (josojo) [inline shared-arguments/src/lib.rs:16]: The local host as default would be nice for testing

**Review by josojo — APPROVED** [review]
nice, was just about to prepare the same

File: shared-arguments/src/lib.rs (hunk 1 above)
Comment (e00E) [reply shared-arguments/src/lib.rs]: Changed default to `http://localhost:8545`.

Comment (fleupold) [reply shared-arguments/src/lib.rs]: Not a big deal but personally I find the node_url shouldn't have a default as it's the one thing you should deliberately think about and specify when running this thing (what network am I on).

**Pull request merged**